enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,295 @@
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+ The CNI will not set an address at virt-launcher pod netns, that address is
+ assigned to the VM with the DHCP options from the LSP, this allows to use 
+ kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The /32 routing feature allows an address to be running at a node different from
+the one "owning" the subnet the address is coming from. This will happen
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what ip pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement only applies for hypershift kubevirt workers so the rest of 
+virtual machine should not use it, those VMs will be marked with label 
+`kubevirt.io/allow-pod-network-bridge-binding-live-migration` by hypershift
+kubevirt provider and ovn-k will do /32 routing if detected, the of non
+hypershift VM should not add that special label.
+
+There is also a moment where routing need to do the "switch", meaning that the
+vm has to be routed over the host is running on, to detect that the label
+"kubevirt.io/nodeName" can be check since it appears just after live migration
+has finish. Also to discard old virt-launcher pod the creation timestamp can
+be used, ignoring them is they are not the latest one.
+
+### Risks and Mitigations
+
+### OVN extended arp_proxy features ####
+
+
+The current ovn arp_proxy is missing the following features: enforcing the mac 
+address and ipv6, ideally those get implemented and ovn-kubernetes have 
+them for 4.14 release.
+
+In case they don't arrive on time, it will not be able to configure IPv6 hosted
+clusters at hypershift and VMs arp table will need to be refreshed after live
+migration. 
+
+### Open Questions 
+
+Do this impact non hypershift payloads ? The control-plane will have some more
+load from the /32 routing but data-plane is not affected
+
+Do this work with future interconnect topology ? It's not using a distributed 
+switch per hosted cluster so that part should be fine, but it may needed 
+to replicate /32 routing at all node's ovn_cluster_router.
+
+Do this need extra configuration at openshift virtualization ? The bridge 
+binding pod network live migration feature gate need to be activated
```

Comment:
is this an open question ? You have a perfect answer for it IMO :)

Reply:
I have follow how it's done at this enhancement and looks like they have also an answer 
https://github.com/openshift/enhancements/blob/master/enhancements/network/ovn-on-hypershift.md#open-questions

Reply:
Hm, on the link you've pasted **some** of them do ... others don't have an answer at all. Some other defer the answer to follow-ups. One even says performance must be taken into account.

I'm OK if you want to leave it as is, but I feel if you know the answer to question, they are not "open questions", and that info belongs elsewhere in the docs - probably structured in a different way. 

Not a must, you call.

Reply:
I will kept them so we don't forget.